Add redispatch API

FyzHsn opened the pull request:
Fixes #1512 

<!--
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Increment the version number in the /VERSION file. If this is a bugfix, increment the patch version (the rightmost number), for example 0.18.2 becomes 0.18.3. If it's a new feature, increment the minor version (the middle number), for example 0.18.2 becomes 0.19.0.
⚠️ Add a note to /CHANGELOG.md with your version number and the date, summarizing the changes.
⚠️ If your pull request fixes an open issue, please link to the issue.
⚠️ Rebase the latest changes from the develop branch. Assuming origin points to https://github.com/AgnostiqHQ/covalent, you should run git rebase origin/develop.
-->

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

santoshkumarradha reviewed (changes requested):
Hey @FyzHsn awesome write up !!

few small glitches and additions 
- in https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch, can you mention that with redispatch, user can do all this 
```
Redefining particular tasks in the workflow.
Reusing previously executed results as much as possible.
Re-executing the workflow with different set of arguments.
```
without actually having the lattice object, i.e. access to the original experiment as redispatch just uses runid 

- https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch for some reason `.. note::` does not seem to be respected in the build

FyzHsn commented:
> Hey @FyzHsn awesome write up !!
> 
> few small glitches and additions
> 
> * in https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch, can you mention that with redispatch, user can do all this
> 
> ```
> Redefining particular tasks in the workflow.
> Reusing previously executed results as much as possible.
> Re-executing the workflow with different set of arguments.
> ```
> 
> without actually having the lattice object, i.e. access to the original experiment as redispatch just uses runid
> 
> * https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch for some reason `.. note::` does not seem to be respected in the build

@santoshkumarradha Thank you for the corrections! I've addressed the comments. I'm guessing `.. note::` does not work bcs the concepts page is `.md` and not `.rst`.

santoshkumarradha commented:
> > Hey @FyzHsn awesome write up !!
> > few small glitches and additions
> > 
> > * in https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch, can you mention that with redispatch, user can do all this
> > 
> > ```
> > Redefining particular tasks in the workflow.
> > Reusing previously executed results as much as possible.
> > Re-executing the workflow with different set of arguments.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > without actually having the lattice object, i.e. access to the original experiment as redispatch just uses runid
> > 
> > * https://covalent--1513.org.readthedocs.build/en/1513/concepts/basics.html#dispatch for some reason `.. note::` does not seem to be respected in the build
> 
> @santoshkumarradha Thank you for the corrections! I've addressed the comments. I'm guessing `.. note::` does not work bcs the concepts page is `.md` and not `.rst`.

Correct ! it becomes this in markdown - https://pradyunsg.me/furo/reference/admonitions/#basic-usage
